python/ray/train/_internal/stats.py

```diff
@@ -0,0 +1,117 @@
+import logging
+import os
+import threading
+from typing import Dict, Optional
+
+import ray
+from ray.data import Dataset
+from ray.train._internal.schema import TrainDatasetInfo, TrainRunInfo, TrainWorkerInfo
+from ray.train._internal.utils import check_for_failure
+from ray.train._internal.worker_group import WorkerGroup
+
+logger = logging.getLogger(__name__)
+
+
+@ray.remote(num_cpus=0)
+class TrainStatsActor:
+    def __init__(self):
+        self.train_runs = dict()
+
+    def register_train_run(self, run_info: TrainRunInfo):
+        # Register a new train run.
+        self.train_runs[run_info.id] = run_info
```

Comment:
nit: Are we okay if this overrides existing run if it gets called multiple times?